Question 2
Mriganko Roy
20A2HP415

Importing the required libraries for executing the problem


In [1]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
%matplotlib inline

Importing the data

In [2]:
pos_sal = pd.read_csv('https://raw.githubusercontent.com/mrigankoml/MLAP_Term-IV/main/MLAP2021Exam-main/MALP2021%20Question%20paper/Position_Salaries.csv')

Reading the first 5 values

In [3]:
pos_sal.head(5)

Position  Level  Salary
0   Business Analyst      1   45000
1  Junior Consultant      2   50000
2  Senior Consultant      3   60000
3            Manager      4   80000
4    Country Manager      5  110000

Including dummy variables

In [4]:
cat = ['Position']
pos_sal_encoded = pd.get_dummies(pos_sal, columns = cat)

Splitting into X and Y 


In [5]:
X = pos_sal_encoded.drop(columns = 'Salary')
y = pos_sal["Salary"]

Splitting into Train and Test data

In [6]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

Feature Scaling

In [7]:
from sklearn.preprocessing import StandardScaler

ss = StandardScaler()
X_train = ss.fit_transform(X_train)
X_test = ss.transform(X_test)

Training the algorithm

In [8]:
from sklearn.ensemble import RandomForestRegressor
regressor = RandomForestRegressor(n_estimators=10, random_state=0)
regressor.fit(X_train, y_train)
y_pred = regressor.predict(X_test)

Evaluation Metrics

In [9]:
from sklearn import metrics
print('Mean Absolute Error:', metrics.mean_absolute_error(y_test, y_pred))
print('Mean Squared Error:', metrics.mean_squared_error(y_test, y_pred))
print('Root Mean Squared Error:', np.sqrt(metrics.mean_squared_error(y_test, y_pred)))

Mean Absolute Error: 59750.0
Mean Squared Error: 4812625000.0
Root Mean Squared Error: 69373.08555917056


**Under observation, we came to know that the accuracy has not been good in the model. With Mean Absolute Error's value coming out to be 59750, we can say that the methods which we have applied for the model haven't given us a good model.**